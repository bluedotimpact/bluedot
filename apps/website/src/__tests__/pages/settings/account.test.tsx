import { render } from '@testing-library/react';
import {
  describe, expect, test, vi, beforeEach, type Mock,
} from 'vitest';
import useAxios from 'axios-hooks';
import AccountSettingsPage from '../../../pages/settings/account';
import { TrpcWrapper } from '../../trpcWrapper';

// Mock dependencies
vi.mock('axios-hooks');

// Only mock withAuth from @bluedot/ui since it wraps the component with auth logic
// withAuth is a HOC that normally provides auth props from a zustand store
// For testing, we bypass it to directly test the component with mock auth
vi.mock('@bluedot/ui', async () => {
  const actual = await vi.importActual('@bluedot/ui');
  return {
    ...actual,
    withAuth: (Component: React.ComponentType<{ auth: { token: string } }>) => Component,
  };
});

const mockedUseAxios = useAxios as unknown as Mock;

// Mock Next.js router
vi.mock('next/router', () => ({
  useRouter: () => ({
    pathname: '/settings/account',
    push: vi.fn(),
  }),
}));

// Mock Next.js Head component
vi.mock('next/head', () => ({
  default: ({ children }: { children: React.ReactNode }) => children,
}));

describe('AccountSettingsPage', () => {
  const mockUserData = {
    user: {
      name: 'John Doe',
      email: 'john.doe@example.com',
    },
  };

  beforeEach(() => {
    vi.clearAllMocks();

    // Default mock for useAxios to return user data
    mockedUseAxios.mockReturnValue([{
      data: mockUserData,
      loading: false,
      error: null,
    }]);
  });

  test('should render account settings page correctly', () => {
    // The component expects an auth prop
    const mockAuth = { token: 'test-token' };

    // Cast the component to accept auth prop since we mocked withAuth
    const AccountSettingsPageWithAuth = AccountSettingsPage as React.ComponentType<{ auth: { token: string } }>;

    const { container } = render(<AccountSettingsPageWithAuth auth={mockAuth} />, { wrapper: TrpcWrapper });

    expect(container).toMatchSnapshot();
  });
});
