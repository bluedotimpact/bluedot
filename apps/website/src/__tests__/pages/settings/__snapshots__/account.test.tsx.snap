// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AccountSettingsPage > should render account settings page correctly 1`] = `
<div>
  <div>
    <title>
      Account | BlueDot Impact
    </title>
    <div
      class="breadcrumbs bg-color-canvas border-b border-color-divider w-full py-space-between"
    >
      <nav
        aria-label="Breadcrumbs"
        class="breadcrumbs__nav section-base flex flex-row justify-between"
      >
        <ol
          class="breadcrumbs__list flex"
        >
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="bluedot-a not-prose breadcrumbs__link no-underline"
              href="/"
              tabindex="0"
            >
              Home
            </a>
            <span
              aria-hidden="true"
              class="breadcrumbs__separator mx-2"
            >
              &gt;
            </span>
          </li>
          <li
            class="breadcrumbs__item flex items-center"
          >
            <p
              class="bluedot-p not-prose breadcrumbs__link !m-0"
            >
              Settings
            </p>
            <span
              aria-hidden="true"
              class="breadcrumbs__separator mx-2"
            >
              &gt;
            </span>
          </li>
          <li
            class="breadcrumbs__item flex items-center"
          >
            <a
              class="bluedot-a not-prose breadcrumbs__link no-underline opacity-50"
              href="/settings/account"
              tabindex="0"
            >
              Account
            </a>
          </li>
        </ol>
      </nav>
    </div>
    <div
      class="container mx-auto px-4 py-6"
    >
      <div
        class="lg:grid lg:grid-cols-12 lg:gap-8"
      >
        <aside
          aria-label="Settings sidebar"
          class="lg:col-span-3 mb-6 lg:max-w-[320px]"
        >
          <h2
            class="bluedot-h2 not-prose mb-4 font-semibold text-size-lg leading-[1.4]"
          >
            Settings
          </h2>
          <nav
            aria-label="Settings navigation"
          >
            <ul
              class="flex gap-2 flex-wrap lg:block lg:space-y-1"
            >
              <li>
                <a
                  aria-label="Account settings"
                  class="
                  w-auto lg:w-full h-11 px-4 py-3 rounded-lg transition-colors flex items-center
                  bg-[rgba(0,85,255,0.08)] text-bluedot-darker font-semibold
                "
                  href="/settings/account"
                  tabindex="0"
                >
                  <span
                    class="flex items-center gap-3 w-full"
                  >
                    <p
                      class="bluedot-p not-prose text-[13px] leading-[22px] flex-grow !m-0 font-semibold"
                    >
                      Account
                    </p>
                  </span>
                </a>
              </li>
              <li>
                <a
                  aria-label="Courses settings"
                  class="
                  w-auto lg:w-full h-11 px-4 py-3 rounded-lg transition-colors flex items-center
                  text-bluedot-darker hover:bg-[rgba(0,85,255,0.04)]
                "
                  href="/settings/courses"
                  tabindex="0"
                >
                  <span
                    class="flex items-center gap-3 w-full"
                  >
                    <p
                      class="bluedot-p not-prose text-[13px] leading-[22px] flex-grow !m-0 font-normal"
                    >
                      Courses
                    </p>
                  </span>
                </a>
              </li>
            </ul>
          </nav>
        </aside>
        <main
          aria-label="Settings content"
          class="lg:col-span-9"
        >
          <div
            role="region"
          >
            <div
              class="p-8"
            >
              <div
                class="mb-6"
              >
                <div
                  class="flex flex-col gap-2"
                >
                  <div
                    id="profile-name-label"
                  >
                    <p
                      class="bluedot-p not-prose font-semibold"
                    >
                      Profile Name*
                    </p>
                  </div>
                  <div
                    class="flex gap-2 items-center"
                  >
                    <label
                      class="input flex flex-col gap-2 flex-1 min-w-0"
                    >
                      <span
                        class="input__label"
                      />
                      <input
                        aria-describedby="profile-name-label"
                        aria-invalid="false"
                        aria-label="Profile name"
                        class="input--text border border-color-divider rounded-lg p-4 focus-visible:outline-2 focus-visible:-outline-offset-2 focus-visible:outline-bluedot-normal flex-1"
                        placeholder="Enter your name"
                        type="text"
                        value="John Doe"
                      />
                    </label>
                  </div>
                </div>
              </div>
              <div
                class="border-t border-color-divider my-6"
              />
              <div
                class="mb-6"
              >
                <p
                  class="bluedot-p not-prose font-semibold mb-2"
                >
                  Email*
                </p>
                <p
                  class="bluedot-p not-prose text-gray-600"
                >
                  john.doe@example.com
                </p>
              </div>
              <div
                class="border-t border-color-divider my-6"
              />
              <div
                class="mb-6"
              >
                <p
                  class="bluedot-p not-prose font-semibold mb-2"
                >
                  Password
                </p>
                <button
                  aria-label="Change password"
                  class="cta-button flex items-center justify-center transition-all duration-200 w-fit whitespace-nowrap cursor-pointer not-prose disabled:opacity-50 disabled:pointer-events-none aria-disabled:opacity-50 aria-disabled:pointer-events-none text-sm px-4 py-3 rounded-sm font-[650] cta-button--secondary bg-transparent border border-bluedot-normal text-bluedot-normal hover:bg-bluedot-lighter"
                  type="button"
                >
                  Change Password
                </button>
              </div>
            </div>
          </div>
        </main>
      </div>
    </div>
  </div>
</div>
`;
